### PR TITLE
Cython Bindings / Array View + Tree View

### PR DESCRIPTION
**Pull request by mahnerak**
* Typed arrays,
* Removed some annotations:
  * Generics are not supported by Cython,
  * `@abstractmethod`s stop us from using `cdef`-ed classes as interfaces,
* Moved implementations to a separate file: Now `arrayview` is only for interface.
* Cython declarations for `TreeArrayView`, `ArrayView`, `TreeView`. Compiling them with Cython.

### CODE REVIEW

File: aim/storage/treearrayview.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+from itertools import islice
+
+from typing import Any, Iterator, List, Tuple, Union
+from aim.storage.treeview import TreeView
+
+from aim.storage.arrayview import ArrayView
+
+
+class TreeArrayView(ArrayView):
```
Comment (alberttorosyan) [inline aim/storage/treearrayview.py:11]: need to change signature for `ProxyTree` as well (for remote tracking).

File: aim/sdk/run.py [hunk 2]
```diff
@@ -645,9 +645,9 @@ def _read_sequence_info(self, ctx, name):
         seq_info = self.sequence_info[sequence_selector]
 
         assert not seq_info.initialized
-        seq_info.val_view = self.series_run_tree.subtree(sequence_selector).array('val').allocate()
-        seq_info.epoch_view = self.series_run_tree.subtree(sequence_selector).array('epoch').allocate()
-        seq_info.time_view = self.series_run_tree.subtree(sequence_selector).array('time').allocate()
+        seq_info.val_view = self.series_run_tree.subtree(sequence_selector).array('val', dtype='double').allocate()
```
Comment (alberttorosyan) [inline aim/sdk/run.py:648]: this function is meant for generic sequences. we cannot be sure that `dtype='double'` is the correct choice.
though looking into the implementation of `TreeArrayView` it is only used to convert to numpy...

Comment (mahnerak) [reply aim/sdk/run.py:648]: This is my fault. Yes, we have to figure out how to treat series / sequences as typed arrays.
Will roll back this one.

**Review by alberttorosyan — APPROVED** [review]
lgtm

**Pull request merged**